mne/event.py

```diff
@@ -745,14 +745,21 @@ def make_fixed_length_events(raw, id, start=0, stop=None, duration=1.):
         of the recording.
     duration: float
         The duration to separate events by.
+    first_samp: bool
+        If True (default), times will have raw.first_samp added to them,
+        as in :func:`mne.find_events`. This behavior is not desirable if 
+        the returned events will be combined with event times that already 
+        have ``raw.first_samp`` added to them, e.g. event times that come 
+        from :func:`mne.find_events`.
 
     Returns
     -------
     new_events : array
         The new events.
     """
     start = raw.time_as_index(start)
-    start = start[0] + raw.first_samp
+    if first_samp:
+        start = start[0] + raw.first_samp
     if stop is not None:
         stop = raw.time_as_index(stop)
         stop = min([stop[0] + raw.first_samp, raw.last_samp + 1])
```

Comment:
Yes, good catch. And then since raw.last_samp is always adjusted for raw.first_samp, do we also need to adjust the comparison of stop when first_samp = False?

Something like:
```python
start = raw.time_as_index(start)
if stop is not None:
    stop = raw.time_as_index(stop)
else:
    stop = raw.last_samp + 1    
if first_samp:
    start = start[0] + raw.first_samp
    stop = min([stop[0] + raw.first_samp, raw.last_samp + 1])  # default behavior for stop
else:
    stop = min([stop[0], raw.last_samp - raw.first_samp])
```
?